tools/Azure.Mcp.Tools.AppLens/src/Commands/Resource/ResourceDiagnoseCommand.cs

```diff
@@ -56,18 +53,19 @@ Once proper input is provided this tool returns a list of insights and solutions
     protected override void RegisterOptions(Command command)
     {
         base.RegisterOptions(command);
-        command.Options.Add(_questionOption);
-        command.Options.Add(_resourceOption);
-        command.Options.Add(_resourceTypeOption);
-        RequireResourceGroup();
+        command.Options.Add(OptionDefinitions.Common.ResourceGroup.AsRequired());
+        command.Options.Add(AppLensOptionDefinitions.Question);
+        command.Options.Add(AppLensOptionDefinitions.Resource);
+        command.Options.Add(AppLensOptionDefinitions.ResourceType);
     }
 
     protected override ResourceDiagnoseOptions BindOptions(ParseResult parseResult)
     {
         var options = base.BindOptions(parseResult);
-        options.Question = parseResult.GetValueOrDefault(_questionOption) ?? string.Empty;
-        options.Resource = parseResult.GetValueOrDefault(_resourceOption) ?? string.Empty;
-        options.ResourceType = parseResult.GetValueOrDefault(_resourceTypeOption) ?? string.Empty;
+        options.ResourceGroup ??= parseResult.GetValueOrDefault<string>(OptionDefinitions.Common.ResourceGroup.Name);
+        options.Question = parseResult.GetValueOrDefault<string>(AppLensOptionDefinitions.Question.Name) ?? string.Empty;
+        options.Resource = parseResult.GetValueOrDefault<string>(AppLensOptionDefinitions.Resource.Name) ?? string.Empty;
```

Comment:
Feels like having a `GetValueOrElse` type extension would be useful

```suggestion
        options.Question = parseResult.GetValueOrElse(AppLensOptionDefinitions.Question.Name, string.Empty);
        options.Resource = parseResult.GetValueOrElse(AppLensOptionDefinitions.Resource.Name, string.Empty);
```

Reply:
I like the idea, but I'm also maxed out on this PR, so will make a note.